fxa-oauth-server/lib/routes/token.js

```diff
@@ -72,11 +72,11 @@ const PAYLOAD_SCHEMA = Joi.object({
     .default(MAX_TTL_S)
     .optional(),
 
-  scope: Joi.alternatives().when('grant_type', {
-    is: GRANT_REFRESH_TOKEN,
-    then: validators.scope,
-    otherwise: Joi.optional()
-  }),
```

Comment:
I couldn't quite parse what this use of `Joi.alternatives()` was trying to do, so I've re-worked it to what I think it *should* be doing, in the same style as other validators in the same file.  The tests pass, so I guess that means it's still doing the right thing?

Reply:
Not sure if I understand correctly either, but is it possible to add a test case that the old behaviour is now prohibited at the API level?

Along the lines of, make a request with a scope and some other `grant_type`, then check that the status code is 400 or whatever?

Reply:
👍 added